src/types.rs

```diff
@@ -39,6 +45,23 @@ impl<'a> core::iter::Iterator for PropertiesIter<'a> {
                 Some(Ok(next_item))
             }
 
+            Properties::CorrelatedSlice {
+                correlation,
+                properties,
+            } => {
+                if self.index == 0 {
+                    self.index += 1;
+                    return Some(Ok(*correlation));
+                }
+
+                let index = self.index - 1;
+                if index >= properties.len() {
+                    return None;
+                }
+
+                Some(Ok(properties[index]))
+            }
```

Comment:
If we want to simplify iterating across properties, we can easily just only do an `IntoIter` implementation for `Properties::DataBlock` and raise `unimplemented!()` otherwise. I don't see any foreseeable case where a user would iterate across `Properties::Slice` or `Properties::CorrelatedSlice`